File: 657.robot-return-to-origin/solution.kt
class Solution {
    fun judgeCircle(moves: String): Boolean {
        val (r, c) = moves
            .fold(0 to 0) { (r, c), m ->
                when (m) { 
                    'R' -> r to c + 1
                    'L' -> r to c - 1
                    'U' -> r - 1 to c
                    'D' -> r + 1 to c
                    else -> throw IllegalArgumentException()
                }
            }
    
        return r == 0 && c == 0
    }
}

File: 657.robot-return-to-origin/solution.go
func judgeCircle(moves string) bool {
    r := 0
    c := 0
    for _, m := range moves {
        switch m {
        case 'R': c++
        case 'L': c--
        case 'U': r--
        case 'D': r++
        }
    }
    return r == 0 && c == 0
}